api/errors/daemon.go

```diff
@@ -0,0 +1,21 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/daemon component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeNoSuchContainer is generated when we look for a container by
+	// name or ID and we can't find it.
+	ErrorCodeNoSuchContainer = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "NOSUCHCONTAINER",
+		Message:        "no such id: %s",
```

Comment:
@cpuguy83 yes we can make those types of changes but I'd like to do that after the initial pass of just moving the error messages we have today into this package/format. We can get into a lot of bike shedding while trying to twiddle error messages and consolidate them, and so I'd like to do that in subsequent PRs. My main goal of this PR is to have zero UX impact and keep all changes internal - and to not break anything/anyone.